File: bocoree/BKeyEventArgs.cs
/*
 * BKeyEventArgs.cs
 * Copyright (c) 2009 kbinani
 *
 * This file is part of bocoree.
 *
 * bocoree is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * bocoree is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
#if JAVA
package org.kbinani.windows.forms;

import java.awt.event.*;
import org.kbinani.*;

public class BKeyEventArgs extends BEventArgs{
    private KeyEvent m_original = null;

    public BKeyEventArgs( KeyEvent e ){
        m_original = e;
    }

    public char getKeyChar(){
        reutrn e.getKeyChar();
    }

    public int getKeyCode(){
        return e.getKeyCode();
    }

    public int getModifiers(){
        return e.getModifiers():
    }
}
#endif

File: org.kbinani.windows.forms/BKeyEventArgs.java
package com.github.cadencii.windows.forms;

import java.awt.event.KeyEvent;
import org.kbinani.BEventArgs;

public class BKeyEventArgs extends BEventArgs{
    private KeyEvent m_original = null;
    public boolean Alt;
    public boolean Control;
    public int KeyValue;
    public boolean Shift;
    public int KeyCode;
    
    public BKeyEventArgs( KeyEvent e ){
        m_original = e;
        
        Alt = m_original.isAltDown();
        Control = m_original.isControlDown();
        KeyValue = m_original.getKeyCode();
        Shift = m_original.isShiftDown();
        KeyCode = m_original.getKeyCode();
    }
    
    public char getKeyChar(){
        if( m_original == null ){
            return '\0';
        }else{
            return m_original.getKeyChar();
        }
    }
    
    public int getKeyCode(){
        if( m_original == null ){
            return 0;
        }else{
            return m_original.getKeyCode();
        }
    }
    
    public int getModifiers(){
        if( m_original == null ){
            return 0;
        }else{
            return m_original.getModifiers();
        }
    }
}
